### ISSUE TITLE
actionMap error

### ISSUE THREAD

**Issue by djonatanb** [burzum/cakephp-user-tools#26]
I think that I cannot map actions from xxxxxx.com/users/request_password to action request_password, it goes to action requestPassword instead. With that the action map doesn't seem to work. if I change the action map it works. 
Am I doing something wrong ?

**Comment by burzum** [comment]
I think there has been a change in the dispatcher of the CakePHP framework itself. What version are you using?

**Comment by djonatanb** [comment]
3.1.7

**Comment by burzum** [comment]
So just changing all the methods from underscored to camelCased in the components map fixes the problem for you? I couldn't find anything related to that in the 3.1 changelog and I have three apps using that version without that I've noticed trouble.

**Comment by djonatanb** [comment]
Inside the UserToolComponent
```
'actionMap' => [
			'index' => [
				'method' => 'listing',
				'view' => 'Burzum/UserTools.UserTools/index',
			],
			'register' => [
				'method' => 'register',
				'view' => 'Burzum/UserTools.UserTools/register'
			],
			'login' => [
				'method' => 'login',
				'view' => 'Burzum/UserTools.UserTools/login',
			],
			'logout' => [
				'method' => 'logout',
				'view' => null
			],
			'resetPassword' => [
				'method' => 'resetPassword',
				'view' => 'Burzum/UserTools.UserTools/reset_password',
			],
			'requestPassword' => [
				'method' => 'requestPassword',
				'view' => 'Burzum/UserTools.UserTools/request_password',
			],
			'changePassword' => [
				'method' => 'changePassword',
				'view' => 'Burzum/UserTools.UserTools/change_password',
			],
			'verifyEmail' => [
				'method' => 'verifyEmailToken',
				'view' => 'Burzum/UserTools.UserTools/verify_email',
			],
			'view' => [
				'method' => 'getUser',
				'view' => 'Burzum/UserTools.UserTools/view',
			]
		]
```

in my boostrap

```
Configure::write('UserTools.Component', [
    'actionMap' => [
        'index' => [
            'method' => 'listing',
            'view' => 'UserTools.UserTools/index',
        ],
        'register' => [
            'method' => 'register',
            'view' => 'register'
        ],
        'changePassword' => [
            'method' => 'changePassword',
            'view' => 'change_password'
        ],
        'login' => [
            'method' => 'login',
            'view' => 'login',
        ],
        'logout' => [
            'method' => 'logout',
            'view' => null
        ],
    ],
]);
```

that's how it works for me, I tried to access it like request-password or request_password but the $action in 
```
public function mapAction() {
		$action = $this->request->params['action'];
```
was always returning it camelCased. When I changed it worked.

**Comment by djonatanb** [comment]
isn't it part of the new conventions to have actions camelBacked ?

**Comment by burzum** [comment]
I'm pretty sure it is because of PSR. I just can't see where I have missed that change to the CakePHP core. I've checked the release notes but haven't seen it there. With 3.0 it was working the way it is. I fear there is a difference between old and new apps. So the difference might be in the cakephp/app package, loading another dispatcher filter or something like that.

**Comment by burzum** [comment]
@djonatanb please see

* https://github.com/cakephp/cakephp/pull/8096
* https://github.com/cakephp/cakephp/pull/8094

**Comment by djonatanb** [comment]
Not sure if I understand it all, but for me the problem seemed to be the action name. As far as I understand you are using action names with underscore as the key in the config array.

Are you using dashedRoute in the projects with cakephp3.0 that work ok with the underscore actions ?

**Comment by burzum** [comment]
@djonatanb no need to understand it, I've fixed it in a backward compatible way. Would you please mind switching to the `develop` branch of the plugin for a moment? It's fixed there. After your feedback I'll tag version 1.0.3.

**Comment by djonatanb** [comment]
ok, gotcha, will do that

**Comment by djonatanb** [comment]
The routing is working for me now

**Comment by burzum** [comment]
A PR for what? If you want an user registration you can re-use the token mechanism. Just trigger the email with the verification token when the admin creates the user and disable the user registration.

**Comment by burzum** [comment]
I've merged to master an tagged 1.0.3. Thank you.
